### ISSUE TITLE
License Manager Recipe Fails with Setup Program From My.Esri.Com

### ISSUE THREAD

**Issue by ZianAtFirstWatch** [Esri/arcgis-cookbook#114]
I tried to run the following role after downloading the License Manager installer from my.esri.com:

{
  "arcgis":{
    "version":"10.5",
    "licensemanager":{
      "setup":"C:\\Users\\fwadmin2\\Desktop\\10.5.1\\ArcGIS_License_Manager_Windows_1051_156259.exe"
    }
  },
  "run_list":[
    "recipe[arcgis-desktop::licensemanager]"
  ]
}

However, even though Chef knows all the configuration information, I was still asked to click on a bunch of things ranging from a "Close" button to kick off a 2nd installer to confirming (again) that I wanted License Manager installed to the default directory in %programfiles%.

**Comment by cdflint** [comment]
First you need to specify "setup_archive" for that particular file.  If you unpack that file to the default %USERPROFILE%\Documents directory you would specify that path to ...\LicenseManager\Setup.exe as "setup".

Further issue is with the latest release of 3.3.1 (3.3.0 of arcgis-desktop) getting an error failing to compile the arcgis-desktop::licensemanager recipe at all.  Chef_solo output below.  Using the setup_archive of license manager 2018.1 with arcgis version set to 10.7.  This is being done on a vm on local machine not AWS.

```shell
==> lm:
==> lm: ================================================================================
==> lm: Recipe Compile Error in C:/chef/cache/cookbooks/arcgis-desktop/recipes/licensemanager.rb
==> lm: ================================================================================
==> lm: NoMethodError
==> lm: -------------
==> lm: undefined method `setups_repo' for Custom resource arcgis_desktop_licensemanager from cookbook arcgis-desktop
==> lm: Cookbook Trace:
==> lm: ---------------
==> lm:   C:/chef/cache/cookbooks/arcgis-desktop/recipes/licensemanager.rb:26:in `block in from_file'
==> lm:   C:/chef/cache/cookbooks/arcgis-desktop/recipes/licensemanager.rb:24:in `from_file'
==> lm: Relevant File Content:
==> lm: ----------------------
==> lm: C:/chef/cache/cookbooks/arcgis-desktop/recipes/licensemanager.rb:
==> lm:
==> lm:  19:
==> lm:  20:  arcgis_desktop_licensemanager 'Verify ArcGIS License Manager system requirements' do
==> lm:  21:    action :system
==> lm:  22:  end
==> lm:  23:
==> lm:  24:  arcgis_desktop_licensemanager 'Unpack ArcGIS License Manager Setup' do
==> lm:  25:    setup_archive node['arcgis']['licensemanager']['setup_archive']
==> lm:  26>>   setups_repo node['arcgis']['repository']['setups']
==> lm:  27:    only_if { ::File.exist?(node['arcgis']['licensemanager']['setup_archive']) &&
==> lm:  28:              !::File.exist?(node['arcgis']['licensemanager']['setup']) }
==> lm:  29:    not_if { Utils.product_installed?(node['arcgis']['licensemanager']['product_code']) }
==> lm:  30:    action :unpack
==> lm:  31:  end
==> lm:  32:
==> lm:  33:  arcgis_desktop_licensemanager 'Install ArcGIS License Manager' do
==> lm:  34:    setup node['arcgis']['licensemanager']['setup']
==> lm:  35:    product_code node['arcgis']['licensemanager']['product_code']
==> lm:
==> lm: System Info:
==> lm: ------------
==> lm: chef_version=14.12.9
==> lm: platform=windows
==> lm: platform_version=10.0.17763
==> lm: ruby=ruby 2.5.5p157 (2019-03-15 revision 67260) [x64-mingw32]
==> lm: program_name=C:/opscode/chef/bin/chef-solo
==> lm: executable=C:/opscode/chef/bin/chef-solo
==> lm:
==> lm: Running handlers:
==> lm: [2019-04-30T17:39:28+00:00] ERROR: Running exception handlers
==> lm: Running handlers complete
==> lm: [2019-04-30T17:39:28+00:00] ERROR: Exception handlers complete
==> lm: Chef Client failed. 0 resources updated in 36 seconds
==> lm: [2019-04-30T17:39:28+00:00] FATAL: Stacktrace dumped to C:/chef/cache/chef-stacktrace.out
==> lm: [2019-04-30T17:39:28+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
==> lm: [2019-04-30T17:39:28+00:00] FATAL: NoMethodError: undefined method `setups_repo' for Custom resource arcgis_desktop_licensemanager from cookbook arcgis-desktop
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
```

chef-stacktrace output.

```shell
Generated at 2019-04-30 17:39:28 +0000
NoMethodError: undefined method `setups_repo' for Custom resource arcgis_desktop_licensemanager from cookbook arcgis-desktop
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/resource.rb:1306:in `method_missing'
C:/chef/cache/cookbooks/arcgis-desktop/recipes/licensemanager.rb:26:in `block in from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/resource_builder.rb:66:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/resource_builder.rb:66:in `build'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/dsl/declare_resource.rb:314:in `build_resource'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/dsl/declare_resource.rb:271:in `declare_resource'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/dsl/resources.rb:38:in `arcgis_desktop_licensemanager'
C:/chef/cache/cookbooks/arcgis-desktop/recipes/licensemanager.rb:24:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/cookbook_version.rb:199:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/run_context.rb:350:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:166:in `block in compile_recipes'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:163:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:163:in `compile_recipes'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:79:in `compile'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/run_context.rb:199:in `load'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/policy_builder/expand_node_object.rb:97:in `setup_run_context'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/client.rb:515:in `setup_run_context'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/client.rb:281:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/application.rb:303:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/application.rb:279:in `block in run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/local_mode.rb:44:in `with_server_connectivity'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/application.rb:261:in `run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/application/client.rb:449:in `run_application'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/application.rb:66:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/lib/chef/application/solo.rb:230:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.12.9-universal-mingw32/bin/chef-solo:24:in `<top (required)>'
C:/opscode/chef/bin/chef-solo:75:in `load'
C:/opscode/chef/bin/chef-solo:75:in `<main>'
```

Hope this helps.

**Comment by cameronkroeker** [comment]
This was fixed in cookbook release v3.6.0:
https://github.com/Esri/arcgis-cookbook/releases
